nhsl-gravity-forms-name-formatter.php

```diff
@@ -47,25 +44,25 @@ function validate_name_fields( $result, $value, $form, $field ) {
 
 				// Set the result array
 				$result['is_valid'] = false;
-				$result['message']  = $result['message'] . 'It looks like you have entered invalid characters for your First Name. Use only letters, hyphens, and apostrophes.<br />';
+				$result['message']  = $result['message'] . 'Your First Name doesn\'t look quite right. Your name should start with a capital letter followed by lower case letters and only contain letters, hyphens, and apostrophes.<br />';
 
 			} // End of If - First Name validation
 
 		// If the middle name is not empty and not hidden then validate the input
-		if ( ! empty( $first ) && ! $field->get_input_property( '4', 'isHidden' ) && ! validate_string( $middle ) ) {
+		if ( ! empty( $middle ) && ! $field->get_input_property( '4', 'isHidden' ) && ! validate_string( $middle ) ) {
 
 				// Set the result array
 				$result['is_valid'] = false;
-				$result['message']  = $result['message'] . 'It looks like you have entered invalid characters for your Middle Name. Use only letters, hyphens, and apostrophes.<br />';
+				$result['message']  = $result['message'] . 'Your Middle Name doesn\'t look quite right. Your name should start with a capital letter followed by lower case letters and only conrain letters, hyphens, and apostrophes.<br />';
 
 			} // End of If - Middle Name validation
 
 		// If the last name is not empty and not hidden then validate the input
-		if ( ! empty( $first ) && ! $field->get_input_property( '6', 'isHidden' ) && ! validate_string( $last ) ) {
+		if ( ! empty( $last ) && ! $field->get_input_property( '6', 'isHidden' ) && ! validate_string( $last ) ) {
 
 				// Set the result array
 				$result['is_valid'] = false;
-				$result['message']  = $result['message'] . 'It looks like you have entered invalid characters for your Last Name. Use only letters, hyphens, and apostrophes.<br />';
+				$result['message']  = $result['message'] . 'Your Last Name doesn\'t look quite right. Your name should start with a capital letter followed by lower case letters and only conrain letters, hyphens, and apostrophes.<br />';
```

Comment:
Typo "conrain" plus the statement "You name should start with a capital letter followed by lower case letters" doesn't allow for names such as MacIntyre, O'Connor, McDonald, Smedley-Hughes, etc.
Also, I don't think using percentage of capitals is the way to go. I suggest we expect a capital at the start and then another after any "Mac" or "Mc" or any apostrophes or hyphens. But, to be on the safe side, I suggest we allow the user to override our rules if they really want to. That way, we don't have to allow for every obscure edge case.